### DISCUSSION TITLE
Wireless troubles + advice on using both I2C + SPI

### DISCUSSION THREAD

**Discussion by ahshah** [SpenceKonde/ATTinyCore#812] [Support / Q & A]
I have a project that needs to read information from an I2C sensor and transmit it wirelessly. Currently I get by with a Tiny85 using a 433mhz  transmitter and a generic regenerative receiver. The solution works decently, but its been flakey- I've had some 433mhz boards die inexplicably, and i'm sometimes loosing data. Its also verrrry sensitive to bad quality wallwarts, despite putting a 0.1uf cap across vcc and gnd on the RX 433mhz board. Theres lots of noise on received packets, Finally I can't transmit quickly (+120ms to transmit a 4 byte payload). The data drops are the big deal for my case- maybe switching to a super heterodyne style receiver will fix these issues, if not then I'm thinking of switching to a nRF24L01 based solution. 

the nRF24L01 boards communicate over SPI and so I can't really use a tiny85 along with the USI / I2C for the sensor and the USI / SPI system at the same time. Short of writing a software SPI, I'm thinking of switching logic to another board entirely. The question is: would the 841 board work for this case? On paper it seems so. Would appreciate any advice!

**Comment by hmeijdam** [comment]
I noticed quite an improvement after switching to syperheterodyne RX's. I particulary like the NICERF [SRX882](https://www.nicerf.com/item/superheterodyne-receiver-srx882s).
I used their old version with a brown ceramic resonator, They have an [Aliexpress shop](https://www.aliexpress.com/store/934254/search?SearchText=srx882).

**Reply by SpenceKonde** [reply]
Yeah nicerf is always a cut above the pack. There's an SRX887 now though that's supposed to be better than the 882 (the 882 was the third best receiver  out of like 10 that I tested. #1 was a $50 US made one. #2 was the el cheapo yellow syn470 board, and #3 was the 882. The STX882 was also - by a nose - better than the rest of the pack for transmitters. (the $8 US transmitter was second to last place - only the green superregenerative board was worse (and it *really* *is* *bad*). Like, the numbers came out to like "1000; feet, 1100 feet, 800 feet, then the superregenerative which was marked as "less than" 50 or 100, becasue there wasn't a suitable observation point very close to where the stationary half was in those tests. and as at that distance, no signal was detected. Those green receivers are absolute trash,
